src/cmake-tools.ts

```diff
@@ -252,15 +265,7 @@ export class CMakeTools implements vscode.Disposable, api.CMakeToolsAPI {
       throw new Error(localize('bad.cmake.executable', 'Bad CMake executable "{0}".', cmake.path));
     }
 
-    let workspace = null;
-    const rootPath = getPrimaryWorkspaceFolder();
-    if (!rootPath) {
-      throw new Error(localize('no.workspace.error', 'CMake Tools is not available without an open workspace'));
-    }
-    if (vscode.workspace.workspaceFolders) {
-      workspace = lightNormalizePath(rootPath.fsPath);
-    }
-
+    const workspace = this.folder.uri.fsPath;
```

Comment:
Because previously CMakeTools can exist without a folder. But now a folder is required.

Reply:
Interesting... I thought that wasn't the case as the root folder is always used.... But anyway we don't need to worry about this in the new code. cmaketools can only be accessed via folder. So even if there's an invalid cmaketools instance somehow, it can't be accessed. If there's nothing open. Then nothing would happen for commands related to cmake tools.

Reply:
ok